pkg/reconciler/apis/apiexport/apiexport_reconcile.go

```diff
@@ -211,14 +211,9 @@ func (c *controller) updateVirtualWorkspaceURLs(ctx context.Context, apiExport *
 			apiExport.Name,
 		)
 
-		desiredURLs.Insert(u.String())
-	}
-
-	apiExport.Status.VirtualWorkspaces = nil
-
-	for _, u := range desiredURLs.List() {
 		apiExport.Status.VirtualWorkspaces = append(apiExport.Status.VirtualWorkspaces, apisv1alpha1.VirtualWorkspace{
-			URL: u,
+			URL:    u.String(),
+			Labels: clusterWorkspaceShard.Labels,
```

Comment:
also, please make sure that the output (`apiExport.Status.VirtualWorkspaces)` will be stable (ordered), otherwise we will end up modifying the export on each sync

Reply:
That's a good point. I need to check that